table/block_based/block_based_table_reader.h

```diff
@@ -500,6 +500,8 @@ class BlockBasedTable : public TableReader {
                            InternalIterator* meta_iter,
                            const InternalKeyComparator& internal_comparator,
                            BlockCacheLookupContext* lookup_context);
+  // If index and filter blocks do not need to be pinned, `prefetch_all`
+  // determines whether they will be read and add to cache.
```

Comment:
I guess it will determines whether to attempt to add to cache then, since it can end up not being cached.